scripts/bert/finetune_classifier.py

```diff
@@ -111,60 +142,89 @@ def evaluate():
         Ls.append(ls)
         step_loss += sum([L.asscalar() for L in Ls])
         metric.update([label], [out])
-    logging.info('validation accuracy: %s', metric.get()[1])
+    logging.info('Validation accuracy: {:.3f}'.format(metric.get()[1]))
 
 
 def train():
     """Training function."""
-    trainer = gluon.Trainer(model.collect_params(), args.optimizer,
-                            {'learning_rate': lr, 'epsilon': 1e-9})
+    try:
+        trainer = gluon.Trainer(model.collect_params(), args.optimizer,
+                                {'learning_rate': lr, 'epsilon': 1e-6, 'wd': 0.01})
+    except ValueError as e:
+        warnings.warn("AdamW optimizer is not found. Please consider upgrading to "
+                      "mxnet>=1.5.0. Now the original Adam optimizer is used instead.")
+        trainer = gluon.Trainer(model.collect_params(), 'adam',
+                                {'learning_rate': lr, 'epsilon': 1e-6, 'wd': 0.01})
 
     num_train_examples = len(data_train)
-    num_train_steps = int(num_train_examples / batch_size * args.epochs)
+    step_size = batch_size * accumulate if accumulate else batch_size
+    num_train_steps = int(num_train_examples / step_size * args.epochs)
     warmup_ratio = args.warmup_ratio
     num_warmup_steps = int(num_train_steps * warmup_ratio)
     step_num = 0
-    differentiable_params = []
+    params = []
 
     # Do not apply weight decay on LayerNorm and bias terms
     for _, v in model.collect_params('.*beta|.*gamma|.*bias').items():
         v.wd_mult = 0.0
-
+    # Collect differentiable parameters
     for p in model.collect_params().values():
         if p.grad_req != 'null':
-            differentiable_params.append(p)
+            params.append(p)
+    # Set grad_req if gradient accumulation is required
+    if accumulate:
+        for p in model.collect_params().values():
+            p.grad_req = 'add'
```

Comment:
this seems wrong... it's setting all parameters to add even for those that had `grad_req = 'null'`

Reply:
try
```
params = [p for p in model.collect_params().values() if p.grad_req != 'null']
if accumulate:
    for p in params:
        p.grad_req = 'add'
```

Reply:
You're right. Constant params have grad_req=null and their req cannot be updated. But if user set some grad_req to `null`, those will be overridden. Updated now.